app/models/appointment.rb

```diff
@@ -1,4 +1,12 @@
 class Appointment < ApplicationRecord
   belongs_to :user
   belongs_to :doctor
+
+  # validations
+  # validate :date_of_appointment_cannot_be_in_the_past
+  validates :description, presence: true, length: { in: 3..150 }
+  validates :date_of_appointment, presence: true,
+                                  format: { with: /\A\d{4}-\d{2}-\d{2}\z/, message: 'Date format should be YYYY-MM-DD' }, inclusion: { in: Date.today..Date.today + 5.years, message: 'Date should be in the future' }
```

Comment:
We will have to add unique to appointments date when the user is meeting the same doctor to avoid problems.